pkg/virt-handler/mount-manager/mount.go

```diff
@@ -0,0 +1,112 @@
+package mount_manager
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	containerdisk "kubevirt.io/kubevirt/pkg/container-disk"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	"kubevirt.io/kubevirt/pkg/virt-handler/isolation"
+	container_disk "kubevirt.io/kubevirt/pkg/virt-handler/mount-manager/container-disk"
+	hotplug_volume "kubevirt.io/kubevirt/pkg/virt-handler/mount-manager/hotplug-disk"
+	mountutils "kubevirt.io/kubevirt/pkg/virt-handler/mount-manager/utils"
+)
+
+//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE
+
+// MountInfo wraps all the mount information
+type MountInfo struct {
+	containerDisksInfo map[string]*containerdisk.DiskInfo
+}
+
+func (info *MountInfo) GetContainerDisksInfo() map[string]*containerdisk.DiskInfo {
+	return info.containerDisksInfo
+}
+
+// MountManager handles all the mount operations required by KubeVirt
+type MountManager interface {
+	Mount(vmi *v1.VirtualMachineInstance) (MountInfo, error)
+	Unmount(vmi *v1.VirtualMachineInstance) error
+	MountsReady(vmi *v1.VirtualMachineInstance, notInitializedSince time.Time) (bool, error)
+	SyncMounts(vmi *v1.VirtualMachineInstance) error
+	SyncUnmounts(vmi *v1.VirtualMachineInstance) error
+	IsMounted(vmi *v1.VirtualMachineInstance, volume string, sourceUID types.UID) (bool, error)
+}
+
+type mountManager struct {
+	containerDiskMounter container_disk.Mounter
+	hotplugVolumeMounter hotplug_volume.VolumeMounter
+}
+
+func NewMounter(virtPrivateDir string, podIsolationDetector isolation.PodIsolationDetector, clusterConfig *virtconfig.ClusterConfig, kubeletPodsDir string) MountManager {
+	return &mountManager{
+		containerDiskMounter: container_disk.NewMounter(podIsolationDetector, clusterConfig, mountutils.NewMountRecorder(filepath.Join(virtPrivateDir, "container-disk-mount-state"))),
+		hotplugVolumeMounter: hotplug_volume.NewVolumeMounter(mountutils.NewMountRecorder(filepath.Join(virtPrivateDir, "hotplug-volume-mount-state")), kubeletPodsDir),
+	}
+}
+
+// MountsReady returns if the mount points are ready to be used
+func (m *mountManager) MountsReady(vmi *v1.VirtualMachineInstance, notInitializedSince time.Time) (bool, error) {
```

Comment:
I would make the naming a bit more explicit here:
```suggestion
func (m *mountManager) ContainerDiskMountsReady(vmi *v1.VirtualMachineInstance, notInitializedSince time.Time) (bool, error) {
```

Reply:
Same explanation as above, I did it on purpose to keep the naming generic

Reply:
I changed the naming